pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -360,20 +382,109 @@ func findOperation(cmd *cobra.Command) (operation, error) {
 	}
 }
 
+func getVmiVerbosity(currentLv map[string]uint, vm *v1.VirtualMachine, vmNamespace string) error {
+	if *vm.Spec.Running {
+		// check pod object's environmental variable VIRT_LAUNCHER_LOG_VERBOSITY
+		podList, err := virtClient.CoreV1().Pods(vmNamespace).List(context.Background(), k8smetav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, pod := range podList.Items {
+			if pod.GenerateName != "virt-launcher-"+vmiName+"-" {
+				continue
+			}
+			for _, container := range pod.Spec.Containers {
+				if container.Name != "compute" {
+					continue
+				}
+				for _, env := range container.Env {
+					if env.Name == "VIRT_LAUNCHER_LOG_VERBOSITY" {
+						currentLv["vmi"], err = atou(env.Value)
+						if err != nil {
+							return err
+						}
+						break
+					}
+					// if there is no VIRT_LAUNCHER_LOG_VERBOSITY, the virt-launcher verbosity was set as default
+				}
+			}
```

Comment:
If you return here, then we can avoid the else branch and the nesting is a bit simpler

Reply:
Do you mean I should do the early return (i.e. return here instead of break)?
```
for _, env := range container.Env {
	if env.Name == "VIRT_LAUNCHER_LOG_VERBOSITY" {
		currentLv["vmi"], err = atou(env.Value)
		if err != nil {
			return err
		}
		return nil
	}
	// if there is no VIRT_LAUNCHER_LOG_VERBOSITY, the virt-launcher verbosity was set as default
}
```

Reply:
Not I mean to return before the `else` so you can avoid it

Reply:
Else block is about the case when VM is not running. So, I think we need the else block even if we return before else block.
- In case of VM is running, we check the launcher pod’s VIRT_LAUNCHER_LOG_VERBOSITY environment variable.
- In case of VM is not running, we check the VM object’s logVerbosity label or the virt-launhcer log verbosity.

Here is the spec. To show the log verbosity of a vmi:
- When the vmi is not running, show the verbosity when the vmi starts running.
   - When logVerbosity label is specified, show the label in the vm object.
   - When logVerbosity label is not specified, show the virt-launcher log verbosity.
- When the vmi is running, show the verbosity of the current running vmi, even if the new verbosity was set after the vmi started running. This is because the new verbosity is applied when the vmi is restarted.
   - Show the pod object's VIRT_LAUNCHER_LOG_VERBOSITY environment variable
   - If VIRT_LAUNCHER_LOG_VERBOSITY does not exist, the virt-launcher verbosity has been set as default, so show the default verbosity (2).